File: src/js/main/RichJsonClassMapping.js
export const __RICH_JSON_CLASS_MAPPING = {}

/**
 * Adds the given class mappings.
 * @param classMappings
 */
export function addClassMappings(classMappings) {
    r
    let names = Object.keys(classMappings);
    let name = undefined;
    for (let i = 0; i < names.length; ++i) {
        name = names[i];
        addClassMapping(name, classMappings[name])
    }
}

/**
 * Adds the given class to the mapping table.
 * @param name
 * @param classType
 */
export function addClassMapping(name, classType) {
    if (Object.hasOwn(__RICH_JSON_CLASS_MAPPING, name)) {
        throw (`RichJson has the class '${name}' already defined`);
    }
    __RICH_JSON_CLASS_MAPPING[name] = classType;
}

export function __mapClassByName(name) {
    if (!Object.hasOwn(__RICH_JSON_CLASS_MAPPING, name)) {
        throw (`RichJson could not find the class called '${name}'.\nMake sure its defined in RichJsonClassMapping.`);
    }
    return __RICH_JSON_CLASS_MAPPING[name];
}

File: src/c++/main/other/RichJsonClassMapping.cpp
#include "RichJsonClassMapping.hpp"
#include <iostream>
#include <stdexcept>

namespace RichJson {

void RichJsonClassMapping::addClassMapping(const std::string& name, Factory factory) {
    if (mapping().find(name) != mapping().end()) {
        std::cerr << "[WARN] RichJson has the class '" << name << "' already defined\n";
        return;
    }
    mapping()[name] = std::move(factory);
}

void RichJsonClassMapping::addClassMappings(const std::unordered_map<std::string, Factory>& mappings) {
    for (const auto& [name, factory] : mappings) {
        addClassMapping(name, factory);
    }
}

const RichJsonClassMapping::Factory& RichJsonClassMapping::mapClassByName(const std::string& name) {
    auto it = mapping().find(name);
    if (it == mapping().end()) {
        throw std::invalid_argument(
            "RichJSON could not find the class called '" + name + "'.\nMake sure its registered via registerClass().");
    }
    return it->second;
}

}
